### PR TITLE
Integration test to verify controller pods status

### PR DESCRIPTION
**Pull request by figo**
The resources are verified by kubectl during apply,
but controller pods are unverified.

also invoke the new tests been adding under ./test/integration/
with this, we should expect the integration ci job automatically
pick up new test under ./test/integration

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

cc @vincepri  @detiber @pablochacin

### CODE REVIEW

**Comment by detiber** [comment]
Checking on the output of the integration tests, I noticed that there is an Error displayed, but the test still shows as passing: https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api/811/pull-cluster-api-integration/102

**Comment by figo** [comment]
> Checking on the output of the integration tests, I noticed that there is an Error displayed, but the test still shows as passing: https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api/811/pull-cluster-api-integration/102

@detiber the error happened when run `bazel run //:gazelle`, re-run succeed, problem is Gazelle return success when encountered issue.
i did some research, looks like a known design of Gazelle, not sure how much we can do about it at his moment, https://groups.google.com/forum/#!topic/bazel-go-discuss/lBMp9Jg7uNE

File: scripts/ci-integration.sh [hunk 1]
```diff
@@ -67,12 +68,33 @@ delete_bootstrap() {
    kind delete cluster --name "${BOOTSTRAP_CLUSTER_NAME}"
 }
 
-verify_cluster_api() {
-   echo "not implemented"
-   #TODO: verify cluster-api CRD on bootstrap cluster
+wait_pod_running() {
+   retry=30
+   INTERVAL=6
```
Comment (vincepri) [inline scripts/ci-integration.sh]: Any reason these vars are in different cases?

Comment (figo) [reply scripts/ci-integration.sh]: good catch, no strong reason, thanks

**Pull request merged**